### ISSUE TITLE
Excessive Height in ProfileCourses Screen Header

### ISSUE THREAD

**Issue by darrylsyms** [darrylsyms/fretwise-app#11]
I am unable to use `navigation.state.routeName` prop inside the `indexScreenApiHooks.setHeaderHeight` hook. This prop crashes the app immediately when the ProfileCourses Screen is opened:
```
externalCodeSetup.indexScreenApiHooks.setHeaderHeight((defaultHeaderHeight, filterType, navigation) => {

	if (filterType === "courses" && navigation.state.routeName === "ProfileCourses" ) {
		return 140
	}

	return 150;
});
```

I have tested `console.log(navigation.state.routeName)`, and the result is most certainly `ProfileCourses`:

<img width="309" alt="Capture 2022-03-06 at 19 15 07@2x" src="https://user-images.githubusercontent.com/82924984/156916935-0eda1e01-b2e5-447f-930a-139b1048de65.png">

**I can confirm that the `filterType` prop works. It is `navigation.state.routeName` that causes this problem.**

## What I'm trying to fix:
There is a large gap in the header of the ProfileCourses Screen after using the `indexScreenApiHooks.setHeaderHeight` hook. This can be easily rectified by using the `navigation.state.routeName` prop from react-navigation, but unfortunately this prop crashes the app when the ProfileCourses Screen is opened. 

![Capture 2022-03-02 at 13 22 16](https://user-images.githubusercontent.com/82924984/156288959-bb26e7f0-7fa8-4021-a87e-9c8b09b3682a.png)

**Comment by darrylsyms** [comment]
#9

**Comment by darrylsyms** [comment]
I also don't like how it appears here:

![IMG_0A5079DE6B1D-1](https://user-images.githubusercontent.com/82924984/152268558-87caa251-e419-45e7-97bf-0ea92196a6f6.jpeg)

I think that AnimatedHeader Components should be removed from all "courses" Screens. However this will also delete the Search Component. The Search Component should be included in the headerRight instead: #23
